staging/src/k8s.io/kubectl/pkg/util/completion/completion.go

```diff
@@ -167,6 +194,26 @@ func CompGetContainers(f cmdutil.Factory, podName string, toComplete string) []s
 	return CompGetFromTemplate(&template, f, "", []string{"pod", podName}, toComplete)
 }
 
+// CompGetContainersPorts gets the list of containers of the specified pod which begin with `toComplete`.
+func CompGetContainersPorts(f cmdutil.Factory, resourceType string, resourceName string, toComplete string) []string {
```

Comment:
*If* my other comments are correct, then I think this function could probably be renamed to something like `CompGetPodContainerPorts` and remove `resourceType` arg, rename `resourceName` to `podName` to be more explicit as to what it is working on.